### DISCUSSION TITLE
Trouble using TSPL2

### DISCUSSION THREAD

**Discussion by planet-fisherman** [php-aidc/label-printer#9] [Q&A]
Hello there,
thank you for providing this tool.
I'm trying to use it with a TSC-TTP245C printer but without any success so far.
The examples you provided are based on Direct Protocoll which of what I understand works for Intermec, right?
Can you provide basic usage examples for use with a TSC printer?

```
use PhpAidc\LabelPrinter\Printer;
use PhpAidc\LabelPrinter\Connector\NetworkConnector;

$printer = new Printer(new NetworkConnector('192.168.x.x'));

\var_dump($printer->ask('_VERSION$'));
```
This one returns an empty string on my TSC printer.

Thank you in advance.

**Comment by jhaoda** [comment]
I don't know much about TSC printers. `_VERSION$` is a variable. Probably you should use `OUT` BASIC command to return its value.

**Reply by planet-fisherman** [reply]
Thank you for your reply. Also with the OUT command I'm unable to return any info from the printer. But I'm also unable to follow any other of the  examples. How can I have a most simple label print on a TSC-printer? All the code examples throw errors. At which point do I decide what kind of printer I'm talking to respectively in which language it needs to be adressed? Is there any other syntax refernce for this tool?

**Reply by jhaoda** [reply]
> All the code examples throw errors

Show error.

**Reply by planet-fisherman** [reply]
my PHP-File:
```
<?php
require 'vendor/autoload.php';

use PhpAidc\LabelPrinter\Enum\Unit;
use PhpAidc\LabelPrinter\Enum\Anchor;
use PhpAidc\LabelPrinter\Enum\Charset;
use PhpAidc\LabelPrinter\Printer;
use PhpAidc\LabelPrinter\Compiler;
use PhpAidc\LabelPrinter\Label\Label;
use PhpAidc\LabelPrinter\Label\Element;
use PhpAidc\LabelPrinter\Connector\NetworkConnector;

$label = Label::create(Unit::MM(), 43, 25)
    ->charset(Charset::UTF8())
    ->add(Element::barcode(10, 10, '123456', 'CODE93')->height(60))
;

(new Printer(new NetworkConnector('192.168.100.40')))->print($label);

?>
```

the error thrown:
Fatal error: Uncaught DomainException in C:\xampp\htdocs\Test\vendor\php-aidc\label-printer\src\Printer.php:37 Stack trace: #0 C:\xampp\htdocs\Test\index.php(18): PhpAidc\LabelPrinter\Printer->print(Object(PhpAidc\LabelPrinter\Label\Label)) #1 {main} thrown in C:\xampp\htdocs\Test\vendor\php-aidc\label-printer\src\Printer.php on line 37

**Reply by jhaoda** [reply]
Try this:
```php
(new Printer(new NetworkConnector('192.168.100.40'), Compiler::create(new Tspl())))->print($label);
```

**Reply by planet-fisherman** [reply]
Thank you very much for your help.
Needed to add
`use PhpAidc\LabelPrinter\Language\Tspl;`
Then your code was able to printed a label. But not with the intended content. It seems to call the last label from the printers cache.
Is there any way I can educate myself about the usage of this tool rather then having to trouble you?

**Reply by xavehoo** [reply]
Hi @planet-fisherman , you can find the solution of this problem by checking my closed issue, where I explain how to do it.
